docs/user-guide/station-firmware-upgrade.md

```diff
@@ -0,0 +1,54 @@
+# Firmware upgrade
+
+The Starterkit offers the functionality of performing a firmware upgrade of the
+Basics Station. This document explains which steps are required to execute such
+an upgrade.
+
+1. When provisioning a concentrator device that will require future firmware
+   upgrades, generate a signature key and store it on the device and in a
+   centralized repository of your choice.
+
+   ```shell title="Example for how to generate a signature key (sig-0.key) with openssl"
+   openssl ecparam -name prime256v1 -genkey | openssl ec -out sig-0.pem
+   openssl ec -in sig-0.pem -pubout -out sig-0.pub
+   openssl ec -in sig-0.pub -inform PEM -outform DER -pubin | tail -c 64 > sig-0.key
+   ```
+
+1. When a firmware upgrade of the Station is needed, generate a digest of the
+   executable file of the upgrade.
+   The digest of an upgrade file `upgrade.sh` can be calculated using the previously
+   generated signature key using the command:
+
+   ```shell
+   openssl dgst -sha512 -sign sig-0.pem update.sh > update.sh.sig-0.sha512
+   ```
+
+1. Retrieve the CRC32 Checksum of the signature key
+
+   ```shell
+   cat sig-0.key | gzip -1 | tail -c 8 | od -t ${1:-u}4 -N 4 -An --endian=little | xargs echo > sig-0.crc
+   ```
+
+1. Use LoRa Device Provisioning CLI tool to trigger the upgrade
+
+   ```powershell
+   dotnet run -- upgrade --deveui <station_eui> --version <version> --file <upgrade_file> --digest <file_digest> --checksum <checksum>
```

Comment:
We are going to include a compiled version of the tool in our release, therefore user should not execute it with "dotnet run" but directly calling the binary

Reply:
Wouldn't that be out of line with the current user guide for the CLI tool, where we instruct the user to use `dotnet run`? https://azure.github.io/iotedge-lorawan-starterkit/dev/tools/device-provisioning/

Reply:
Updated the doc to recommend calling the binary of the CLI directly. I also added a link to the documentation of the CLI tool (which is now also updated).